source/Irrlicht/CIrrDeviceLinux.cpp

```diff
@@ -282,6 +286,72 @@ bool CIrrDeviceLinux::switchToFullscreen()
 }
 
 
+void CIrrDeviceLinux::setupTopLevelXorgWindow()
+{
+#ifdef _IRR_COMPILE_WITH_X11_
+	if (CreationParams.DriverType == video::EDT_NULL)
+		return; // no display and window
+
+	os::Printer::log("Configuring X11-specific top level window properties", ELL_DEBUG);
+
+	// Set application name and class hints. For now name and class are the same.
+	// Note: SDL uses the executable name here (i.e. "minetest").
+	XClassHint *classhint = XAllocClassHint();
+	classhint->res_name = const_cast<char *>("Minetest");
+	classhint->res_class = const_cast<char *>("Minetest");

```

Comment:
Technically speaking, those names do not belong to the framework. However, IrrlichtMt is basically modified to fit to Minetest's needs.

Thing is - I would like to use IrrlichtMt over stock Irrlicht for an unrelated project to benefit from the well-maintained code. However, if keeping it generalized is too much effort, I would understand that decision as well.

Reply:
I have strong doubts about the usefulness of using irrlichtmt for another application, as there will always be annoying breaking changes for you.

But if you do still want to do this, you can also just choose the SDL driver, it takes these names from the executable's filename.

=> It's not worth doing imo, sorry.